public/app/plugins/panel/heatmap/specs/renderer_specs.ts

```diff
@@ -115,8 +114,15 @@ describe('grafanaHeatmap', function () {
           let bucketsData = convertToHeatMap(ctx.series, ctx.data.yBucketSize, ctx.data.xBucketSize, logBase);
           ctx.data.buckets = bucketsData;
 
-          // console.log("bucketsData", bucketsData);
-          // console.log("series", ctrl.panel.yAxis.logBase, ctx.series.length);
+          let cardsData = convertToCards(bucketsData);
+          let maxCardsValue = _.max(_.map(cardsData, 'count'));
```

Comment:
not sure I like that the stats are collected outside convertToCard. We already loop through the data in convertToCards, doing it twice again to find max/min seems unnecessary. Can we do the min/max in convertToCards  and have convertToCards return a structure with the stats and the array?